lib/https.js

```diff
@@ -40,6 +40,12 @@ function Server(opts, requestListener) {
   }
   opts = util._extend({}, opts);
 
+  if (opts && typeof opts.key === 'boolean')
```

Comment:
Line 41 guarantee that `opts` is an object so there is no need to check `opts` here or in line 46.